### PR TITLE
feat: publish gmc updates to connections

### PR DESCRIPTION
**Pull request by CaiWillis**

### CODE REVIEW

File: src/main/java/uk/nhs/hee/tis/revalidation/connection/config/ElasticsearchConfig.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package uk.nhs.hee.tis.revalidation.connection.config;
```
Comment (jayanta2018) [inline src/main/java/uk/nhs/hee/tis/revalidation/connection/config/ElasticsearchConfig.java:1]: Looks like it's a new class. Need to put MIT licence here.

File: src/main/java/uk/nhs/hee/tis/revalidation/connection/entity/GmcDoctor.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package uk.nhs.hee.tis.revalidation.connection.entity;
```
Comment (jayanta2018) [inline src/main/java/uk/nhs/hee/tis/revalidation/connection/entity/GmcDoctor.java:1]: MIT Licence

File: src/main/java/uk/nhs/hee/tis/revalidation/connection/message/receiver/ConnectionMessageReceiver.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
+package uk.nhs.hee.tis.revalidation.connection.message.receiver;
```
Comment (jayanta2018) [inline src/main/java/uk/nhs/hee/tis/revalidation/connection/message/receiver/ConnectionMessageReceiver.java:1]: MIT Licence

File: src/main/java/uk/nhs/hee/tis/revalidation/connection/message/receiver/GmcDoctorMessageReceiver.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
+package uk.nhs.hee.tis.revalidation.connection.message.receiver;
```
Comment (jayanta2018) [inline src/main/java/uk/nhs/hee/tis/revalidation/connection/message/receiver/GmcDoctorMessageReceiver.java:1]: Same here, MIT

File: src/main/java/uk/nhs/hee/tis/revalidation/connection/message/receiver/MessageReceiver.java [hunk 5]
```diff
@@ -0,0 +1,6 @@
+package uk.nhs.hee.tis.revalidation.connection.message.receiver;
```
Comment (jayanta2018) [inline src/main/java/uk/nhs/hee/tis/revalidation/connection/message/receiver/MessageReceiver.java:1]: MIT

File: src/main/java/uk/nhs/hee/tis/revalidation/connection/message/receiver/SyncMessageReceiver.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
+package uk.nhs.hee.tis.revalidation.connection.message.receiver;
```
Comment (jayanta2018) [inline src/main/java/uk/nhs/hee/tis/revalidation/connection/message/receiver/SyncMessageReceiver.java:1]: MIT

File: src/test/java/uk/nhs/hee/tis/revalidation/connection/message/receiver/ConnectionMessageReceiverTest.java [hunk 7]
```diff
@@ -0,0 +1,73 @@
+package uk.nhs.hee.tis.revalidation.connection.message.receiver;
```
Comment (jayanta2018) [inline src/test/java/uk/nhs/hee/tis/revalidation/connection/message/receiver/ConnectionMessageReceiverTest.java:1]: MIT

File: src/test/java/uk/nhs/hee/tis/revalidation/connection/message/receiver/GmcDoctorMessageReceiverTest.java [hunk 8]
```diff
@@ -0,0 +1,131 @@
+package uk.nhs.hee.tis.revalidation.connection.message.receiver;
```
Comment (jayanta2018) [inline src/test/java/uk/nhs/hee/tis/revalidation/connection/message/receiver/GmcDoctorMessageReceiverTest.java:1]: MIT

File: src/test/java/uk/nhs/hee/tis/revalidation/connection/message/receiver/SyncMessageReceiverTest.java [hunk 9]
```diff
@@ -0,0 +1,90 @@
+package uk.nhs.hee.tis.revalidation.connection.message.receiver;
```
Comment (jayanta2018) [inline src/test/java/uk/nhs/hee/tis/revalidation/connection/message/receiver/SyncMessageReceiverTest.java:1]: MIT

File: src/test/java/uk/nhs/hee/tis/revalidation/connection/message/receiver/SyncMessageReceiverTest.java [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+
+  @Test
+  void shouldUpdateConnectionsOnReceiveMessageGetMaster() {
+    when(masterElasticSearchService.findAllScroll()).thenReturn(connectionInfoDtos);
+    syncMessageReceiver.handleMessage("getMaster");
+    verify(elasticSearchIndexUpdateHelper, times(1))
+        .updateElasticSearchIndex(connectionInfoDtos.get(0));
+  }
+
+//  @Test
```
Comment (jayanta2018) [inline src/test/java/uk/nhs/hee/tis/revalidation/connection/message/receiver/SyncMessageReceiverTest.java:76]: Test annontation is commented out here?

**Review by jayanta2018 — CHANGES_REQUESTED** [review]

**Review by jayanta2018 — DISMISSED** [review]
LGTM now

**Review by DorisWongNHS — DISMISSED** [review]

**Review by DorisWongNHS — APPROVED** [review]

**Pull request merged**